### DISCUSSION TITLE
Mid lesson 9 challenge

### DISCUSSION THREAD

**Discussion by Abelaby** [Cyfrin/foundry-full-course-cu#324] [Q&A]
Hey, we were asked to implement two test ourselves and check the repo to see if they are right, upon checking the repo, I saw that we  are again advised to implement the testCheckUpkeepReturnsFalseIfEnoughTimeHasntPassed() function, the following is my solution and it passed the test successfully, can anyone confirm that this is the correct way to implement this test?

```solidity
  function testCheckUpkeepReturnsFalseIfEnoughTimeHasntPassed()public{
        vm.prank(PLAYER);
        raffle.enterRaffle{value: entranceFee}();
        vm.warp(block.timestamp);

        (bool upKeepNeeded, ) = raffle.checkUpKeep("");
        assert(!upKeepNeeded);

    }
```

**Comment by ghost** [comment]
Is a valid way to do it but in my opinion you don't need to use the `warp` cheatcode because you're not moving in time.

Another alternative would be to increase the time `block.timestamp + intertval`, increase the block number and still `checkUpkeep` should be false.

**Reply by Abelaby** [reply]
Ty mate

**Reply by cromewar** [reply]
Nice approach @Abelaby, there is not a only way to solve the same problem so the test for this specific problem could be potentially infinite, you did a great job!.

**Comment by DukeLurkin** [comment]
I am so confused.  Isn't interval the duration of the lottery in seconds?  So adding the lottery duration number of seconds to the block does NOT pass time?  Shouldn't the upkeep take place after every interval?

**Reply by tobezhanabi** [reply]
no, it should be within the interval
